tests/load/test_read_interfaces.py

```diff
@@ -0,0 +1,405 @@
+    tables = list(table_relationship.iter_arrow(chunk_size=chunk_size))
+    assert [t.num_rows for t in tables] == expected_chunk_counts
+
+    # check all items are present
+    ids = reduce(lambda a, b: a + b, [t.column(expected_columns[0]).to_pylist() for t in tables])
+    assert set(ids) == set(range(total_records))
+
+    # check fetch accessors
+    table_relationship = pipeline._dataset().items
+
+    # check accessing one item
+    one = table_relationship.fetchone()
+    assert one[0] in range(total_records)
+
+    # check fetchall
+    fall = table_relationship.fetchall()
+    assert len(fall) == total_records
+    assert {item[0] for item in fall} == set(range(total_records))
+
+    # check fetchmany
+    many = table_relationship.fetchmany(chunk_size)
+    assert len(many) == chunk_size
+
+    # check iterfetchmany
+    chunks = list(table_relationship.iter_fetch(chunk_size=chunk_size))
+    assert [len(chunk) for chunk in chunks] == expected_chunk_counts
+    ids = reduce(lambda a, b: a + b, [[item[0] for item in chunk] for chunk in chunks])
+    assert set(ids) == set(range(total_records))
+
+    # check that hints are carried over to arrow table
+    expected_decimal_precision = 10
+    expected_decimal_precision_2 = 12
+    if destination_config.destination_type == "bigquery":
+        # bigquery does not allow precision configuration..
+        expected_decimal_precision = 38
+        expected_decimal_precision_2 = 38
+    assert (
+        table_relationship.arrow().schema.field("decimal").type.precision
+        == expected_decimal_precision
+    )
+    assert (
+        table_relationship.arrow().schema.field("other_decimal").type.precision
+        == expected_decimal_precision_2
+    )
+
+    # simple check that query also works
+    tname = pipeline.sql_client().make_qualified_table_name("items")
+    query_relationship = pipeline._dataset()(f"select * from {tname} where id < 20")
+
+    # we selected the first 20
+    table = query_relationship.arrow()
+    assert table.num_rows == 20
+
+    # check join query
+    tdname = pipeline.sql_client().make_qualified_table_name("double_items")
+    query = (
+        f"SELECT i.id, di.double_id FROM {tname} as i JOIN {tdname} as di ON (i.id = di.id) WHERE"
+        " i.id < 20 ORDER BY i.id ASC"
+    )
+    join_relationship = pipeline._dataset()(query)
+    table = join_relationship.fetchall()
+    assert len(table) == 20
+    assert list(table[0]) == [0, 0]
+    assert list(table[5]) == [5, 10]
+    assert list(table[10]) == [10, 20]
+
+    # special filesystem sql checks
+    if destination_config.destination_type == "filesystem":
+        import duckdb
+        from dlt.destinations.impl.filesystem.sql_client import FilesystemSqlClient
+
+        # check we can create new tables from the views
+        with pipeline.sql_client() as c:
+            c.create_views_for_tables({"items": "items", "double_items": "double_items"})
+            c.execute_sql(
```

Comment:
why do you need to register views explicitly? there's a subquery referencing those tables so you should be able to parse them out with sqlglot no? (I just checked and you get list of all tables from this expression so this line is not needed)

Reply:
I remove it; i think this was a leftover from another query.